android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -117,6 +120,47 @@ public void onDismiss() {
         setUpViews();
     }
 
+    public void showPopupMenu(){
+       PopupMenu menu = new PopupMenu(mOptionsImage.getContext(),   (View) mOptionsImage);
+       menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+       menu.setOnMenuItemClickListener(item -> {
+            if (item.getItemId() == R.id.action_lock_wallet) {
+                mBraveWalletPanelServices.getKeyringService().lock();
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_connected_sites) {
+                //TODO needs to be implemented in another ticket
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_settings) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    activity.openBraveSettings();
+                    dismiss();
+                }
+            }
+            if (item.getItemId() == R.id.action_view_on_block_explorer) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    mBraveWalletPanelServices.getKeyringService().getKeyringInfo(
+                    BraveWalletConstants.DEFAULT_KEYRING_ID, keyringInfo -> {
+                        if (keyringInfo != null) {
+                            mAccountInfos = keyringInfo.accountInfos;
+                            if (mAccountInfos.length > 0)
+                                activity.viewOnBlockExplorer(mAccountInfos[0].address);
```

Comment:
Can directly use the `Utils.openAddress("/address/"+address, mJsonRpcService, this);` here, no need to add `viewOnBlockExplorer` helper method in the BraveActivity.

Reply:
tried this but it does not work while `PopupWindow` is alive and we have called `dismiss()`. In this scenario `Utils.openAddress()` is not working

Reply:
Yeah, it's probably because the `mJsonRpcService` is closed when the popup is dismissed, let's leave this as it is for now, though what are your thoughts on using a domain layer vs the current approach? would love to have more architecture improvements in codebase, any thoughts?

Reply:
Yeah its better have a domain layer vs current approach. It's better that we follow that pattern i,e Domain layer, LiveData etc moving forward with new code. Later on we can migrate previous code too.

Reply:
That's the plan though more updates are coming on the domain layer, just adding some changes to support different chains or coins with my upcoming PR, at least for the time being, would love to see how it can be used to address the current limitations and future challenges.